pkg/virt-launcher/virtwrap/network/network.go

```diff
@@ -35,116 +29,69 @@ import (
-	podInterfaceNames := mapPodInterfaceNameByNetwork(primaryNet, secondaryNets)
-	handler := &NetworkUtilsHandler{}
-	for i, iface := range vmi.Spec.Domain.Devices.Interfaces {
-		network, ok := networks[iface.Name]
-		if !ok {
-			return fmt.Errorf("failed to find a network %s", iface.Name)
-		}
-		podnic := podNICFactory(handler, cacheFactory)
-		podInterfaceName := podInterfaceNames[network.Name]
-		err := podNIC.PlugPhase1(podnic, vmi, &vmi.Spec.Domain.Devices.Interfaces[i], &network, podInterfaceName, pid)
-		if err != nil {
-			return err
-		}
+func newVMNetworkConfiguratorWithHandlerAndCache(vmi *v1.VirtualMachineInstance, handler NetworkHandler, cacheFactory cache.InterfaceCacheFactory) *VMNetworkConfigurator {
+	return &VMNetworkConfigurator{
+		vmi:          vmi,
+		handler:      handler,
+		cacheFactory: cacheFactory,
 	}
-	return nil
 }
 
-func SetupPodNetworkPhase2(vmi *v1.VirtualMachineInstance, domain *api.Domain, cacheFactory cache.InterfaceCacheFactory) error {
-	networks := mapNetworksByName(vmi.Spec.Networks)
-	primaryNet := lookupPrimaryNetwork(vmi.Spec.Networks)
-	secondaryNets := filterSecondaryMultusNetworks(vmi.Spec.Networks)
-	podInterfaceNames := mapPodInterfaceNameByNetwork(primaryNet, secondaryNets)
-	handler := &NetworkUtilsHandler{}
-	for i, iface := range vmi.Spec.Domain.Devices.Interfaces {
-		network, ok := networks[iface.Name]
-		if !ok {
-			return fmt.Errorf("failed to find a network %s", iface.Name)
-		}
-		podnic := podNICFactory(handler, cacheFactory)
-		podInterfaceName := podInterfaceNames[network.Name]
-		err := podNIC.PlugPhase2(podnic, vmi, &vmi.Spec.Domain.Devices.Interfaces[i], &network, domain, podInterfaceName)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
+func NewVMNetworkConfigurator(vmi *v1.VirtualMachineInstance, cacheFactory cache.InterfaceCacheFactory) *VMNetworkConfigurator {
+	return newVMNetworkConfiguratorWithHandlerAndCache(vmi, &NetworkUtilsHandler{}, cacheFactory)
 }
 
-func mapNetworksByName(nets []v1.Network) map[string]v1.Network {
-	networks := map[string]v1.Network{}
-	for _, net := range nets {
-		networks[net.Name] = net
-	}
-	return networks
+func (v VMNetworkConfigurator) getNICs() ([]podNIC, error) {
+	return v.getNICsWithLauncherPID(nil)
+
 }
 
-func mapPodInterfaceNameByNetwork(primaryMultusNetwork *v1.Network, secondaryMultusNetworks []v1.Network) map[string]string {
-	m := map[string]string{}
-	for i, net := range secondaryMultusNetworks {
-		m[net.Name] = getSecondaryPodInterfaceName(i)
+func (v VMNetworkConfigurator) getNICsWithLauncherPID(launcherPID *int) ([]podNIC, error) {
+	nics := []podNIC{}
+
+	if len(v.vmi.Spec.Domain.Devices.Interfaces) == 0 {
+		return nics, nil
 	}
-	if primaryMultusNetwork != nil {
-		m[primaryMultusNetwork.Name] = primaryPodInterfaceName
+
+	for i, _ := range v.vmi.Spec.Networks {
+		nic, err := newPodNIC(v.vmi, &v.vmi.Spec.Networks[i], v.handler, v.cacheFactory, launcherPID)
```

Comment:
Since we use the network to generate the nic name I choose to iterate those and trait interfaces as a sub product.

Reply:
OK, fair enough.

It just makes this part of the code a lot different than before, and I don't think it was required. Since it is all in the same  commit, it becomes distracting, and gets harder to review.